### PR TITLE
Add Azure IPI install

### PR DESCRIPTION
**Pull request by akrzos**
WIP thus DNM, however please review when time is available.

### CODE REVIEW

**Comment by akrzos** [comment]
@chaitanyaenr  @jmencak PTAL this should be ready

File: OCP-4.X/roles/cleanup-on-azure/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+#
+# Clean up Azure IPI install
+#
+
+- name: cleanup azure cluster
```
Comment (jmencak) [inline OCP-4.X/roles/cleanup-on-azure/tasks/main.yml]: `s/cleanup azure cluster/Cleanup Azure cluster/` ?

File: OCP-4.X/roles/install-on-azure/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      - golang-bin
+
+- name: Use installer from a build
+  block:
+    - name: Extract openshift-install binary from the payload
+      shell: |
+        set -o pipefail
+        cd {{ansible_user_dir}}/scale-ci-deploy/bin
+        oc adm release extract --tools {{openshift_install_release_image_override}}
+        ls *.tar.gz | xargs -I % sh -c 'tar -xvf %'
```
Comment (jmencak) [inline OCP-4.X/roles/install-on-azure/tasks/main.yml]: I'd probably use `find -name \*.tar.gz -exec tar -xvf {} \;` but that's a matter of taste.

File: OCP-4.X/roles/install-on-azure/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        chmod +x openshift-install
+  when: not openshift_install_installer_from_source|bool
+
+- name: Build installer from source
+  block:
+    - name: create gopath dir
```
Comment (jmencak) [inline OCP-4.X/roles/install-on-azure/tasks/main.yml]: `s/create gopath dir/Create gopath dir/`? Similarly, below 4x and elsewhere many times.

File: OCP-4.X/roles/post-config-on-azure/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+- name: Create network security group rule for ssh (22)
+  shell: |
+    az network nsg rule create -g {{rhcos_cluster_name.stdout}}-rg --name scale-ci-ssh --nsg-name {{azure_sg.stdout}} --priority 101 --access Allow --description "scale-ci allow ssh" --protocol Tcp --destination-port-ranges "22"
+
+- name: Create network security group rule for pbench-agent ssh (2022)
+  shell: |
+    az network nsg rule create -g {{rhcos_cluster_name.stdout}}-rg --name scale-ci-pbench-agent --nsg-name {{azure_sg.stdout}} --priority 102 --access Allow --description "scale-ci allow pbench-agent" --protocol Tcp --destination-port-ranges "2022"
+
+- name: Create network security group rule for http tests (9090)
```
Comment (jmencak) [inline OCP-4.X/roles/post-config-on-azure/tasks/main.yml]: This is no longer needed -- let's keep things clean.

File: OCP-4.X/roles/post-config-on-azure/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
+---
+#
+# Post-Config for OCP on Azure
+#
+# Performs:
+# * Opens firewall rules for workload node for ssh (as well as for debugging workload pod ssh)
+# * Opens firewall rules for Network Tests P2P HostNetwork Tests and HTTP tests
+#   Ports 2022 (tcp), 9090 (tcp), 20000-20109 (tcp, udp), 32768 60999 (tcp, udp)
```
Comment (jmencak) [inline OCP-4.X/roles/post-config-on-azure/tasks/main.yml]: Remove reference to 9090.

**Comment by akrzos** [comment]
@jmencak PTAL again

**Comment by jmencak** [comment]
Didn't test this version, but LGTM.  Thanks!  @chaitanyaenr, I think this is ready to merge unless you have objections.

**Pull request merged**